ui/page/wallet/wallet_settings_page.go

```diff
@@ -359,6 +361,61 @@ func (pg *SettingsPage) subSectionSwitch(title string, option *cryptomaterial.Sw
 }
 
 func (pg *SettingsPage) changeSpendingPasswordModal() {
+	var currentPassword, dexPass string
+	// New wallet password modal.
+	newSpendingPasswordModal := modal.NewCreatePasswordModal(pg.Load).
+		Title(values.String(values.StrChangeSpendingPass)).
+		EnableName(false).
+		PasswordHint(values.String(values.StrNewSpendingPassword)).
+		ConfirmPasswordHint(values.String(values.StrConfirmNewSpendingPassword)).
+		SetPositiveButtonCallback(func(walletName, newPassword string, m *modal.CreatePasswordModal) bool {
+			err := pg.wallet.ChangePrivatePassphraseForWallet(currentPassword,
+				newPassword, sharedW.PassphraseTypePass)
+			if err != nil {
+				m.SetError(err.Error())
+				m.SetLoading(false)
+				return false
+			}
+
+			if dexPass != "" { // update wallet password in dex
+				fmt.Println("Got here")
+				assetID, _ := dex.BipSymbolID(pg.wallet.GetAssetType().ToStringLower())
+				err := pg.AssetsManager.DexClient().SetWalletPassword([]byte(dexPass), assetID, []byte(newPassword))
+				if err != nil {
+					m.SetError(fmt.Errorf("Failed to update your dex wallet password, try again: %v", err).Error())
+					m.SetLoading(false)
+					return false
+				}
+			}
+
+			info := modal.NewSuccessModal(pg.Load, values.StringF(values.StrSpendingPasswordUpdated),
+				modal.DefaultClickFunc())
+			pg.ParentWindow().ShowModal(info)
+			return true
+		})
+	pg.ParentWindow().ShowModal(newSpendingPasswordModal)
```

Comment:
Wouldn't this display the `newSpendingPasswordModal` immediately, instead of the `currentPasswordModal` first?